test/support/utils_invoke.h

```diff
@@ -253,6 +253,47 @@ struct invoke_on_all_policies
     }
 };
 
+////////////////////////////////////////////////////////////////////////////////
+template <typename TSourceDataType, typename TSourceDataSize>
+struct test_through_permutation_iterator
+{
+    TSourceDataType src_data;
+    const TSourceDataSize src_data_size = 0;
+    ::std::vector<TSourceDataSize> indexes;
+
+    using perm_it_type = decltype(dpl::make_permutation_iterator(::std::declval<TSourceDataType>(), indexes.begin()));
+
+    test_through_permutation_iterator(TSourceDataType src_data, const TSourceDataSize n)
+        : src_data(src_data), src_data_size(n)
+    {
+    }
+
+    template <typename TStepFunctor, typename Operand>
+    void
+    operator()(TStepFunctor stepOp, Operand op)
+    {
+        for (TSourceDataSize perm_idx_step = 1; perm_idx_step < src_data_size; perm_idx_step = stepOp(perm_idx_step))
+        {
+            TSourceDataSize src_data_idx = 0;
+            for (TSourceDataSize perm_idx = 0; perm_idx < src_data_size; perm_idx += perm_idx_step, ++src_data_idx)
```

Comment:
Probably, it makes sense to write it simpler? without empty loop?
```
TSourceDataSize src_data_idx = src_data_size;
TSourceDataSize perm_idx = perm_idx_step*src_data_size;

```

Reply:
Fixed